### ISSUE TITLE
Consider changing the default/preferred maze file format

### ISSUE THREAD

**Issue by Peque** [mackorone/mms#4]
I would suggest using the https://github.com/micromouseonline/mazefiles format. For the following reasons:

- They are easy to understand by humans.
- They are easy to read/write with any programming language.
- They are easy to visualize (just use your browser or favorite text editor).
- No need for special tools if you want to edit them or create your own.

Also, using that format would allow you to access the huge amount of competition and non-competition maze files available in the repository. :blush:

**Comment by mackorone** [comment]
Thanks for the feedback! I agree with all of your points except for the second. In fact, the main reason why I didn't use that format initially is because it's non-trivial to parse. That being said, I agree that it'd be useful, and in fact I've already got some Python logic that does the job: https://github.com/mackorone/maze/blob/master/formats/map.py. I'll take a stab at re-implementing that logic within the simulator. You don't happen to have a name for the format, do you? If so, I'll use it when updating the README.

**Comment by mackorone** [comment]
I added logic to parse "map" files in b4cab1728e66ac85d0200ed6bb01a1c7b8ece560. Currently, mms only supports goals at the center of the maze, so the logic ignores the "G" characters. Fixing this would require extending the Mouse API to include methods for querying for the goal. I'll save that for a later date. Also, I hope you don't mind, I added a link to your repo in my README.

**Comment by Peque** [comment]
I like to call that format the "standard" micromuse maze file format. I do think it is the most standard out there. For what I know (I may be wrong though):

- No other format is more widely used among contestants. At least not among contestants sharing their code/simulators/maze files.
- No other format is more readable for humans (I prefer to make things easier for humans than to machines).
- It is the only format that has [open discussions on possible format changes](https://github.com/micromouseonline/mazefiles/issues/7) (i.e.: there is not just one person deciding and anybody is welcome to raise their voice).
- At least 3 "professional" (that have participated in the All Japan) micromouse contestants have used/contributed to this repository.
- The repo has the biggest and most complete collection of micromouse competition mazes.
- It is the only micromouse maze file repository that enforces and verifies (through tests and continuous integration) the correct formatting of all mazes.

When I started [my own simulator](https://www.youtube.com/watch?v=6Om6xij6LAw) I did not want to create yet-another-format so I spent some time searching and finally joined what I thought it was the most standard/widely used/convenient format.

I am hoping for this format to be more widely adopted. Thanks for joining the train! :blush:
